crates/sui-core/src/authority.rs

```diff
@@ -2940,8 +2946,10 @@ impl AuthorityState {
             .try_as_package_mut()
             .expect("Created as package");
 
-        let cur_normalized = cur_pkg.normalize().expect("Normalize existing package");
-        let mut new_normalized = new_pkg.normalize().ok()?;
+        let cur_normalized = cur_pkg
+            .normalize(max_binary_format_version)
+            .expect("Normalize existing package");
```

Comment:
In general we can't expect this to succeed, because technically the max binary format version could go down?  Not sure if that's something that would practically happen or that we would even want to support though.

Reply:
The max binary format version cannot go down, because then things on chain won't load? 
But I think either way, we might just want to flip this to an invariant violation instead of an unwrap/expect

Reply:
I am not sure what should we do here then exactly but turning it into a similar patter used to handle other non-panicking errors here - log an error message and return `None` as a result.